iFixit/footer
# Create footer

This PR is the first release of `@ifixit/footer` package.

## QA

Ensure that the package components look like the [mockup](https://ifixit-design.webflow.io/footer) and that everything is exported correctly using the following steps:

1. Clone this repo and build the package with `npm run build` and `npm pack`
2. Create a new Create React App using `npx create-react-app my-app`
3. Copy `ifixit-footer-0.1.0.tgz`  into `my-app` 
4. Install required dependencies: `npm i styled-components @core-ds/primitives ifixit-footer-0.1.0.tgz`
5. Copy `src/stores.json` , `src/links.json`  and `src/partners.json`  into` my-app/src`
6. Copy the following code to `my-app/src/App.js` and ensure that everything compiles/renders correctly

<details>
  <summary>App.js</summary>

```javascript
import { fontWeight, space } from "@core-ds/primitives";
import {
  Divider,
  DividerOrientation,
  FacebookLogo,
  Footer,
  FooterList,
  FooterListItem,
  FooterPartner,
  Globe,
  Grid,
  GridItem,
  InstagramLogo,
  Link,
  RepairOrgLogo,
  Stack,
  StoreMenu,
  TwitterLogo,
  YoutubeLogo,
} from "@ifixit/footer";
import * as React from "react";
import links from "./links.json";
import partners from "./partners.json";
import stores from "./stores.json";

function App() {
  return (
    <div className="App">
      <Footer>
        <Grid
          templateColumns={{
            base: "1fr",
            sm: "repeat(3, 1fr)",
            lg: "repeat(4, 1fr)",
          }}
          gap={space[4]}
          paddingX={{
            base: space[5],
            sm: space[7],
          }}
          paddingY={space[7]}
        >
          {links.map((list, index) => (
            <FooterList key={index}>
              {list.map((link, linkIndex) => (
                <FooterListItem key={linkIndex}>
                  <Link href={link.url} fontWeight={fontWeight.bold}>
                    {link.label}
                  </Link>
                </FooterListItem>
              ))}
            </FooterList>
          ))}
          <FooterList>
            <FooterListItem>
              <Link href="#" prefix={<FacebookLogo />}>
                Facebook
              </Link>
            </FooterListItem>
            <FooterListItem>
              <Link href="#" prefix={<TwitterLogo />}>
                Twitter
              </Link>
            </FooterListItem>
            <FooterListItem>
              <Link href="#" prefix={<InstagramLogo />}>
                Instagram
              </Link>
            </FooterListItem>
            <FooterListItem>
              <Link href="#" prefix={<YoutubeLogo />}>
                Youtube
              </Link>
            </FooterListItem>
            <FooterListItem>
              <Link href="#" prefix={<RepairOrgLogo />}>
                Repair.org
              </Link>
            </FooterListItem>
          </FooterList>
          <GridItem
            columnEnd={{ sm: "span 3", lg: "auto" }}
            marginTop={{ base: space[6], lg: "unset" }}
            justify="center"
          >
            <Grid templateColumns="repeat(3, 1fr)" gap={space[4]}>
              {partners.map((partner, partnerIndex) => (
                <Link href="#" key={partnerIndex}>
                  <FooterPartner alt={partner.name} source={partner.logo} />
                </Link>
              ))}
            </Grid>
          </GridItem>
        </Grid>
        <Divider />
        <Grid
          templateColumns={{ base: "1fr", lg: "repeat(2, 1fr)" }}
          gap={space[4]}
          paddingX={{ base: space[5], sm: space[7] }}
          paddingY={space[5]}
        >
          <GridItem>
            <Stack
              direction={{ base: "column", sm: "row" }}
              spacing={{ base: space[2], sm: space[8] }}
            >
              <StoreMenu
                initialCountryCode="US"
                label="Region"
                onChange={(store) => console.log("Clicked", store.name)}
                stores={stores}
              />
              <Link href="#" suffix={<Globe />}>
                Help translate
              </Link>
            </Stack>
          </GridItem>
          <GridItem justify={{ base: "start", lg: "end" }} align="center">
            <Stack
              align="stretch"
              direction={{ base: "column", sm: "row" }}
              spacing={{ base: space[2], sm: space[4] }}
            >
              <Link href="#">Licensed under Creative Commons</Link>
              <Divider
                orientation={DividerOrientation.Vertical}
                display={{ base: "none", sm: "block" }}
              />
              <Link href="#">Privacy</Link>
              <Divider
                orientation={DividerOrientation.Vertical}
                display={{ base: "none", sm: "block" }}
              />
              <Link href="#">Legal</Link>
            </Stack>
          </GridItem>
        </Grid>
      </Footer>
    </div>
  );
}

export default App;
```
</details>

Additionally, ensure that the store menu is accessible (you should be able to select store using arrow keys).

> Note: Language selector is not part of this release as Shopify stores are single language

Connects #1
